apps/calendar/js/bridge.js

```diff
@@ -52,16 +52,14 @@ exports.observeCalendars = function() {
   });
 
   calendarStore.on('add', getAllAndWrite);
-  calendarStore.on('preRemove', getAllAndWrite);
   calendarStore.on('remove', getAllAndWrite);
   calendarStore.on('update', getAllAndWrite);
 
   stream.cancel = function() {
     calendarStore.off('add', getAllAndWrite);
-    calendarStore.off('preRemove', getAllAndWrite);
     calendarStore.off('remove', getAllAndWrite);
     calendarStore.off('update', getAllAndWrite);
-    stream._emitter.removeAllListeners();
+    stream._cancel();
```

Comment:
I created a temporary method since we will need the same logic in multiple places.